snuba/state/cache/redis/scripts/set.lua

```diff
@@ -11,22 +20,22 @@
 -- responsible for executing. If it doesn't exist or does not match the current
 -- task ID, we must have overrun the timeout.
 -- TODO: This may still be able to safely set the cache value?
-local task_id = redis.call('GET', KEYS[3])
-if not task_id or task_id ~= ARGV[1] then
+local task_id = redis.call('GET', task_id_key)
+if not task_id or task_id ~= task_id then
```

Comment:
Isn't this variable shadowing, having two locals called `task_id`? My lua is not great.

Reply:
yes you're right, great catch